adapt/engine.py

```diff
@@ -121,8 +164,18 @@ def register_intent_parser(self, intent_parser):
 
         :raises ValueError on invalid intent
         """
-        if hasattr(intent_parser, 'validate') and callable(intent_parser.validate):
-            self.intent_parsers.append(intent_parser)
+        if hasattr(
+                intent_parser,
+                'validate') and callable(
+                intent_parser.validate):
+            missing_entities = self.trie.checkForMissingEntites(
```

Comment:
Yes,  I was originally thinking of logging a warning but being that this is in Adapt and not in core I felt raising the exception was important to get the info to core.  I still want to log the warning so this solution isn't 100%.